src/e2ee/RustEngine.ts

```diff
@@ -90,7 +90,21 @@ export class RustEngine {
                 break;
             case "joined":
             default:
-            // Default and other cases handled by assignment before switch
+                memberships = ["join"];
+        }
+
+        const members = new Set<UserId>();
+        for (const membership of memberships) {
+            try {
+                (await this.client.getRoomMembersByMembership(roomId, membership))
+                    .map(u => new UserId(u.membershipFor))
+                    .forEach(u => void members.add(u));
```

Comment:
preferably we use a real `for` loop for this, as one-liners mixed in with try/catch is a bit confusing imo

Reply:
also, we shouldn't need a `Set` here as someone can't exist in two states. Appending to an array (using `push(...userIds)`) should be fine?